### DISCUSSION TITLE
i18n tooling

### DISCUSSION THREAD

**Discussion by rashfael** [venueless/venueless#278] [Ideas]
Currently we are translating our user interface with vue-i18n, which uses it's own file format and syntax which can only be authored directly as a text file.

If and when we are involving more collaborators to localize venueless into more languages, this might become cumbersome fast and is inaccessible for non-programmers.

**I think switching to are more well known library, format and syntax like https://www.i18next.com/ would make sense.**

**Additionally, should we start using and offering a tool like weblate to possible localization collaborators?**

And, as a smaller point, how does this work with the venueless license?

**Comment by raphaelm** [comment]
> Additionally, should we start using and offering a tool like weblate to possible localization collaborators?

I think we should either use a free/cheap cloud service or weblate, since I already know how to host weblate.

> And, as a smaller point, how does this work with the venueless license?

Hmm, I think we could just exclude translations file from the license and consider them BSD-licensed or public domain or whatever.

**Comment by rashfael** [comment]
I migrated from vue-i18n to i18next as our i18n library in this [branch](https://github.com/venueless/venueless/tree/i18next).
The locale files themselves stayed the same, but we could think about namespacing some parts of the frontend, like the admin ui, and lazy load translations only for admin users.

**Comment by rashfael** [comment]
We set up weblate to manage translations here: https://translate.pretix.eu/engage/venueless/
